nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexServiceImpl.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2018-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.repository.helm.internal.content.createindex;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.sonatype.goodies.common.ComponentSupport;
+import org.sonatype.nexus.common.collect.NestedAttributesMap;
+import org.sonatype.nexus.repository.Repository;
+import org.sonatype.nexus.repository.content.Asset;
+import org.sonatype.nexus.repository.content.AssetBlob;
+import org.sonatype.nexus.repository.view.Content;
+import org.sonatype.nexus.repository.view.payloads.StringPayload;
+import org.sonatype.repository.helm.internal.AssetKind;
+import org.sonatype.repository.helm.internal.HelmFormat;
+import org.sonatype.repository.helm.internal.content.HelmContentFacet;
+import org.sonatype.repository.helm.internal.metadata.ChartEntry;
+import org.sonatype.repository.helm.internal.metadata.ChartIndex;
+import org.sonatype.repository.helm.internal.util.YamlParser;
+
+import org.joda.time.DateTime;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sonatype.repository.helm.internal.database.HelmProperties.APP_VERSION;
+import static org.sonatype.repository.helm.internal.database.HelmProperties.DESCRIPTION;
+import static org.sonatype.repository.helm.internal.database.HelmProperties.ICON;
+import static org.sonatype.repository.helm.internal.database.HelmProperties.MAINTAINERS;
+import static org.sonatype.repository.helm.internal.database.HelmProperties.NAME;
+import static org.sonatype.repository.helm.internal.database.HelmProperties.SOURCES;
+import static org.sonatype.repository.helm.internal.database.HelmProperties.VERSION;
+
+/**
+ * Build index.yaml file for Helm Hosted
+ *
+ * @since 1.0.15
+ */
+@Named
+@Singleton
+public class CreateIndexServiceImpl
```

Comment:
Could you create a ticket for adding event handlers and put its number in the todo here?